rust/lance-index/src/vector/graph.rs

```diff
@@ -169,7 +169,8 @@ pub trait Graph {
 /// -------
 /// A descending sorted list of ``(dist, node_id)`` pairs.
 ///
-pub(super) fn beam_search(
+/// WARNING: Internal API, do not use it directly.
+pub fn beam_search(
```

Comment:
`Internal API,  API stability is not guaranteed`

I'm not sure if there are macros available to warn against linking to these APIs outside of this project.